packages/mask/src/plugins/Savings/protocols/LDOProtocol.ts

```diff
@@ -0,0 +1,120 @@
+import type Web3 from 'web3'
+import type { AbiItem } from 'web3-utils'
+import { EthereumTokenType, ChainId, getSavingsConstants } from '@masknet/web3-shared-evm'
+import type { Contract } from 'web3-eth-contract'
+import LidoStETHABI from '@masknet/web3-contracts/abis/LidoStETH.json'
+import type BigNumber from 'bignumber.js'
+import type { SavingsNetwork, SavingsProtocol } from '../types'
+
+export interface LidoContract {
+    type: EthereumTokenType
+    chainName: string
+    ldoContract: string
+    stEthContract: string
+}
+
+export const LidoContracts: { [key: number]: LidoContract } = {
+    [ChainId.Mainnet]: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Ethereum',
+        ldoContract: getSavingsConstants(ChainId.Mainnet).LIDO || '',
+        stEthContract: getSavingsConstants(ChainId.Mainnet).LIDO_STETH || '',
+    },
+    [ChainId.Gorli]: {
+        type: EthereumTokenType.ERC20,
+        chainName: 'Gorli',
+        ldoContract: getSavingsConstants(ChainId.Gorli).LIDO || '',
+        stEthContract: getSavingsConstants(ChainId.Gorli).LIDO_STETH || '',
+    },
+}
+
+export class LidoProtocol implements SavingsProtocol {
+    public id = 0
+    public name = 'Lido'
+    public image = 'lido'
+    public base = 'ETH'
+    public pair = 'stETH'
+    public apr = '0.00'
+    public balance = '0.00'
+
+    public availableNetworks: SavingsNetwork[] = [
+        { chainId: ChainId.Mainnet, chainName: 'Ethereum' },
+        { chainId: ChainId.Gorli, chainName: 'Gorli' },
+    ]
+
+    public getContract(chainId: number, web3: Web3): Contract {
+        const contract = new web3.eth.Contract(LidoStETHABI as AbiItem[], LidoContracts[chainId].stEthContract)
```

Comment:
It's not type-safe to create a contract in this way. After making an ABI JSON in the `web3-contracts`, you can run `npm run start` in the CLI. It will create a TypeScript contract definition for the ABI. The compiled definition file contains a class named by the JSON filename. It can be leveraged by the `createContract`:

```ts
import LidoContract from '@masknet/web3-contracts/types/Lido'

const contract = createContract<LidoContract>(web3, address, ABI as AbiItem[])
```